chaco/scales/formatters.py

```diff
@@ -350,15 +354,19 @@ def estimate_width(self, start, end, numlabels=None, char_width=None,
             elif char_width:
                 avg_size = len("%g%g" % (start, end)) / 2.0
                 initial_estimate = round(fill_ratio * char_width / avg_size)
+            else:
```

Comment:
This seems orthogonal to the cleanup, can you describe what the issue is here? If the issue is that `numlabels` and `char_width` should not be both `None`, perhaps a `ValueError("num_labels and char_width should not both be None.")` would be more appropriate.